### ISSUE TITLE
Port Mozilla tests to Higgs

### ISSUE THREAD

**Issue by maximecb** [higgsjs/Higgs#75]
Mozilla have a large base of tests cases for their SunSpider JS engine, see:
http://hg.mozilla.org/mozilla-central/file/d812f80a0f1d/js/src/tests

We would like to port these tests to work with our own test framework ([test-runner.js](https://github.com/maximecb/Higgs/blob/master/source/test-runner.js), which is executed with the `make test` command). This will require cherry-picking tests which are currently applicable to Higgs as some of these will use features which we do not yet support. It will also require changing parts of the tests to use our own test library ([lib/test.js](https://github.com/maximecb/Higgs/blob/master/source/lib/test.js)) instead of their own idiosyncratic assertion functions.

Each ported test should ideally be a standalone JS file which can be run by Higgs directly from the console (e.g.: ./higgs tests/mozilla/mytest.js). The ported tests should go under source/tests/mozilla. [New issues](https://github.com/maximecb/Higgs/wiki/How-to-Contribute#wiki-reporting-and-fixing-bugs) should be opened for the tests that currently fail under Higgs.

Note that it's important to include the license files associated with the tests that get ported.

**Comment by iankronquist** [comment]
Hello again! The test suite you are interested in is called test262 (named after some committee or something). It can be found on Github: https://github.com/tc39/test262. Running it is as easy as:
`./tools/packaging/test262.py --command /path/to/Higgs/source/higgs`. It would be handy to set up a Travis CI post commit hook to run all the tests.

**Comment by maximecb** [comment]
Can you try running it and tell us what results you get? Does it work out of the box at all?

**Comment by iankronquist** [comment]
Some of the tests pass and some do not, I haven't let it run for the full time yet because I'm impatient. I'm trying to get them to run on Travis CI, but the build may time out, we'll see.
There are enough tests that it might be nice to see a fancy dashboard with which suites are passing and failing. I know Mozilla has such a dashboard for B2G, but I don't know about Spidermonkey.

**Comment by iankronquist** [comment]
The travis tests are running on my fork, if you're curious:
https://travis-ci.org/iankronquist/Higgs

**Comment by maximecb** [comment]
Seems a lot of them are failing due to missing libraries. See "undefined base" errors. Would be worth establishing what causes that.

**Comment by iankronquist** [comment]
So the build has stalled, perhaps while running test `15.4.4.15-3-28`.
Fun fact, Travis CI builds will be killed after 50 minutes.
http://docs.travis-ci.com/user/build-configuration/#Job-Timeouts

**Comment by maximecb** [comment]
I don't think we necessarily need to run this on Travis. The point would be to get the tests to run once  and maybe spot some more or less trivial bugs. Then we can open issues for those, create tests of our own and add them to our own test suite.

Right now, something is clearly preventing the tests from even running properly. Likely a missing dependency.
